content/doc/developer/publishing/releasing-cd.adoc

```diff
@@ -0,0 +1,149 @@
+---
+title: Setting up automated plugin release
+layout: developerguide
+references:
+- url: /jep/229
+  title: 'JEP-229: Continuous Delivery of Jenkins Components and Plugins'
+---
+
+== Prerequisites
+
+. Plugin must be 'incrementalized', see link:../../plugin-development/incrementals[incrementals], a new plugin created from the link:https://github.com/jenkinsci/archetypes/[archetypes] will already have this setup.
+. Enabled the continuous delivery flag in link:https://github.com/jenkins-infra/repository-permissions-updater/[repository permission updater] (RPU) for your plugin.
+
+
+[source,yaml]
+----
+cd:
+  enabled: true
+----
+
+Once that has been merged, start checking https://github.com/jenkinsci/your-plugin/settings/secrets/actions and you should soon see `MAVEN_TOKEN` and `MAVEN_USERNAME` appear under Repository secrets.
+
+== Release notes
+
+Next, if you already have Release Drafter configured, remove any tag-template override in `.github/release-drafter.yml`, and delete `.github/workflows/release-drafter.yml` if using GitHub Actions (or remove the app from the repo otherwise). If you have not yet configured Release Drafter, just create `.github/release-drafter.yml` containing only:
+
+[source,yaml]
+----
+_extends: .github
+----
+
+== Adding the GitHub action
+
+Create `.github/workflows/cd.yaml` as a copy of the link:https://github.com/jenkinsci/.github/blob/master/workflow-templates/cd.yaml[cd.yaml template].
+
+If this is your first time using a GitHub action on this project then navigate to the 'Actions' tab 
+in your GitHub repository and enable GitHub actions.
+
+If you have a `.github/dependabot.yml`, it is a good idea to add:
+
+[source,yaml]
+----
+- package-ecosystem: github-actions
+  directory: /
+  schedule:
+    interval: daily
+----
+
+== Pom modifications
+
+For a regular component whose version number is not that meaningful:
+
+[source,diff]
```

Comment:
Maybe we should create a new goal like `incrementalify` (or option to that goal) to do this sort of edit. And/or a flag in `archetypes`?

Reply:
yes that would be great, I was thinking about this at least in context of the IRC bot, is there any reason not to automatically opt new plugins in to the cd.enabled flag? i.e. don't bother adding a flag to Jira in the hosting issue.

Also do we need a new flag in archetypes or can we assume plugins use the is hosted on jenkinsci flag should use this setup?

Reply:
> is there any reason not to automatically opt new plugins in to the cd.enabled flag?

That would be a question for @daniel-beck probably.

> can we assume plugins us[ing] the is hosted on jenkinsci flag should use this setup?

We should probably get some more diverse adoption before making this the default for new plugins.

Reply:
> is there any reason not to automatically opt new plugins in to the cd.enabled flag?

Assuming this is asking "why not set this even if unused":

* This effectively changes the permission model; anyone with commit access to the repo can deploy (even if jumping through hoops to get the token). Right now folks can add whoever they want to GH as committers and it doesn't affect the bits we distribute; `cd.enabled` changes this.
* Every repo adds about 1-2 seconds to the execution time of the RPU run.